File: sample-game/src/main/java/com/github/ummo93/game/ai/AiBehaviourStrategy.java
package com.github.ummo93.game.ai;

import com.github.ummo93.framework.Actor;
import com.github.ummo93.framework.Scene;
import com.github.ummo93.game.Controllable;


public interface AiBehaviourStrategy {
    <T extends Actor & Controllable> void turn(T subject, Actor object, Scene scene);
}

File: sample-game/src/main/kotlin/com/github/ummo93/game/ai/AiBehaviourStrategy.kt
package com.github.ummo93.game.ai

import com.github.ummo93.framework.Actor
import com.github.ummo93.framework.Scene
import com.github.ummo93.game.Controllable

interface AiBehaviourStrategy {
    fun <T> turn(subject: T, target: Actor?, scene: Scene?) where T : Actor?, T : Controllable?
}